datafeed-dapp/contracts/Dex.sol

```diff
@@ -1,147 +1,83 @@
+            collectionPriceInUSD =
+                collectionResult /
+                uint256(pow(collectionPower));
+        }
 
-    /// @notice Returns swap amount
-    /// @dev Calculate swap amount by fetching price from oracle
-    /// @return toAmount to be transferred inexchange of fromAmount
-    function getSwapAmount(
-        uint16 from,
-        uint16 to,
-        uint256 fromAmount
-    ) public view returns (uint256) {
-        uint32 fromTokenResult;
-        uint32 toTokenResult;
-        int8 fromTokenPower;
-        int8 toTokenPower;
-
-        (fromTokenResult, fromTokenPower) = delegator.getResultFromID(from);
-        (toTokenResult, toTokenPower) = delegator.getResultFromID(to);
-
-        uint256 fromTokenPrice = fromTokenResult / uint256(pow(fromTokenPower));
-        uint256 toTokenPrice = toTokenResult / uint256(pow(toTokenPower));
-
-        uint256 toAmount = (fromAmount * fromTokenPrice) / toTokenPrice;
-        return toAmount;
+        uint256 usdAmount = msg.value * collectionPriceInUSD;
+        IERC20(usdToken).approve(address(this), usdAmount);
+        IERC20(usdToken).transferFrom(address(this), msg.sender, usdAmount);
     }
 
-    /// @notice Only admin can update tokenID
-    /// @dev In case of ID's gets updated in delegator, tokenID needs to be updated
-    function updateTokenID(uint16 _id, uint16 _tokenID)
-        public
-        onlyOwner
-        checkID(_id)
-    {
-        _id == 1 ? token1ID = _tokenID : token2ID = _tokenID;
+    function updateTransparentForwarder(
+        address _transparentForwarder
+    ) public onlyOwner {
+        transparentForwarder = ITransparentForwarder(_transparentForwarder);
     }
 
-    /// @notice Only admin can update tokenAddress
-    /// @dev In case of token gets out of balance, new token contract can be deployed and set
-    function updateTokenAddress(uint16 _id, address _tokenAddress)
-        public
-        onlyOwner
-        checkID(_id)
-    {
-        _id == 1 ? token1 = _tokenAddress : token2 = _tokenAddress;
+    function updateUSDToken(address _usdToken) public onlyOwner {
+        usdToken = _usdToken;
     }
 
-    /// @notice Fund user with required token
-    /// @dev Transfer `fundAmount` to msg.sender with
-    function addFunds(uint16 _id) public checkID(_id) {
-        address tokenAddress = _id == 1 ? token1 : token2;
-        IERC20(tokenAddress).approve(address(this), fundAmount);
-        IERC20(tokenAddress).transferFrom(
-            address(this),
-            msg.sender,
-            fundAmount
-        );
+    function updateCollectionNameHash(
+        bytes32 _collectionNameHash
+    ) public onlyOwner {
+        collectionNameHash = _collectionNameHash;
     }
 
     function pow(int8 val) public pure returns (int256) {
         if (val == 0) {
-            return 0;
+            return 1;
         }
-
         int256 result = 1;
 
-        if (val < 0) {
-            result = -1;
+        if (val < -1) {
```

Comment:
it should be less than 0

Reply:
Resolved in [c9f3c35](https://github.com/razor-network/razor-network-examples/pull/5/commits/c9f3c3534e149c4ee881864eededcc5d40e400c7)